apps/services/user-notification/src/app/modules/notifications/notifications.service.ts

```diff
@@ -108,47 +125,81 @@ export class NotificationsService {
           return template
         }
       }
+
       throw new NotFoundException(`Template: ${templateId} not found`)
     } catch {
       throw new NotFoundException(`Template: ${templateId} not found`)
     }
   }
 
-  validateArgCounts(
-    body: CreateHnippNotificationDto,
-    template: HnippTemplate,
-  ): boolean {
-    return body.args.length == template.args.length
+  /**
+   * Checks if the arguments provided in the request body are valid for the template and checks if the number of arguments match
+   */
+  async validate(templateId: string, args: ArgumentDto[]) {
+    const template = await this.getTemplate(templateId)
+
+    if (!this.validateArgCounts(args, template)) {
+      throw new BadRequestException(
+        `Number of arguments doesn't match, template requires ${template.args.length} arguments but ${args.length} were provided`,
+      )
+    }
+
+    const validArgs = this.validateArgs(args, template)
+
+    if (!validArgs.isValid && validArgs.message) {
+      throw new BadRequestException(validArgs.message)
+    }
+  }
+
+  validateArgCounts(args: ArgumentDto[], template: HnippTemplate): boolean {
+    return args.length == template.args.length
   }
 
-  formatArguments(
-    body: CreateHnippNotificationDto,
-    template: HnippTemplate,
-  ): HnippTemplate {
-    if (template.args.length > 0) {
-      const allowedReplaceProperties = [
-        'notificationTitle',
-        'notificationBody',
-        'notificationDataCopy',
-        'clickAction',
-      ]
-      // find {{arg.key}} in string and replace with arg.value
-      const regex = new RegExp(/{{[^{}]*}}/)
+  /**
+   * Checks if the arguments provided in the request body are valid for the template
+   */
+  validateArgs(args: ArgumentDto[], template: HnippTemplate) {
```

Comment:
I would turn this check around and check that at least all the args required by the template are provided. We should just ignore additional args.

Reply:
Hmm don´t know if I understand you. It is checking if the arg length and temp length are the same before
![image](https://github.com/island-is/island.is/assets/112904566/418f66ed-b0ae-42ec-b261-33cf2da3ebd6)

And also this is just old code moved to the service and refactored a little bit